clients/go/cmd/zbctl/internal/commands/root.go

```diff
@@ -166,23 +174,73 @@ func shouldOverwriteEnvVar(cliParam, envVar string) bool {
 	return cliParameterSet || !exists
 }
 
-func parseAddress() (address string, port string) {
-	address = DefaultAddressHost
+func parseAddress() (host string, port string, err error) {
+
+	if shouldUseAddress() {
+		return getHostPortFromAddress()
+	}
+	host, port = getHostPort()
+	return host, port, nil
+}
+
+func parseHostAndPortFromAddress(parsedAddress string) (string, string, error) {
+	host, port, err := net.SplitHostPort(parsedAddress)
+	if err != nil {
+		if strings.Contains(err.Error(), "missing port in address") {
+			host = parsedAddress
+			port = DefaultAddressPort
+			err = nil
+		}
+	}
+	return host, port, err
+}
+
+func shouldUseAddress() bool {
+	_, hostEnvExists := os.LookupEnv(zbc.GatewayHostEnvVar)
+	_, portEnvExists := os.LookupEnv(zbc.GatewayPortEnvVar)
+	if len(hostFlag) == 0 && len(portFlag) == 0 && !portEnvExists && !hostEnvExists {
+		return true
+	}
+	return false
```

Comment:
Can be simplified into
```suggestion
	return len(hostFlag) == 0 && len(portFlag) == 0 && !portEnvExists && !hostEnvExists 
```

Reply:
Done